cloudify_azure/resources/compute/virtualmachine/virtualmachine_utils.py

```diff
@@ -0,0 +1,90 @@
+# #######
+# Copyright (c) 2016-2020 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cloudify import ctx
+
+
+def check_if_configuration_changed(ctx, update_payload, current_vm):
+    for prop in ['location', 'tags', 'plan', 'availability_set',
```

Comment:
Do you think that these properties will be enough? I notice that [there are more possibilities](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/createorupdate).

Reply:
Currently, we work with this API version:
https://github.com/cloudify-cosmo/cloudify-azure-plugin/blob/master/cloudify_azure/constants.py#L46
which has this model:
https://docs.microsoft.com/en-us/python/api/azure-mgmt-compute/azure.mgmt.compute.v2016_03_30.models.virtualmachine?view=azure-python

If we want to get deep into it, in the current package of Azure we use the highest API version for compute is: v2019_12_01 
https://docs.microsoft.com/en-us/python/api/azure-mgmt-compute/azure.mgmt.compute.v2019_12_01.models.virtualmachine?view=azure-python

And as you can see I added the `eviction_policy` and `priority` options for example.

I agree that If we bump the package version we need to update the code to have the new  API  parameters too.